backend/cmd/rewrite/api.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/ansel1/merry/v2"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+)
+
+type response[T any] struct {
+	Data   T
+	Errors []any
+}
+
+type episode struct {
+	Episode struct {
+		Title       string
+		Description string
+		Image       *string
+		Number      *int
+		Season      struct {
+			Title string
+		}
+	}
+}
+
+type season struct {
+	Season struct {
+		Title       string
+		Description string
+		Image       *string
+		Number      int
+		Show        struct {
+			Title string
+		}
+	}
+}
+
+type show struct {
+	Show struct {
+		Title       string
+		Description string
+		Image       *string
+	}
+}
+
+var apiEndpoint = os.Getenv("API_ENDPOINT")
+
+func get[T any](query string, variables map[string]any) *T {
+	body, err := json.Marshal(map[string]any{
+		"query":     query,
+		"variables": variables,
+	})
+
+	req, err := http.NewRequest("POST", apiEndpoint, strings.NewReader(string(body)))
+	if err != nil {
+		return nil
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil
+	}
+
+	resString, _ := io.ReadAll(res.Body)
+
+	if res.StatusCode < 200 || res.StatusCode > 299 {
+		log.Default().Print(merry.New("Error occurred when fetching", merry.WithHTTPCode(res.StatusCode), merry.WithMessage(string(resString))))
+		return nil
+	}
+
+	var r response[T]
+	err = json.Unmarshal(resString, &r)
+	if err != nil || r.Errors != nil {
+		return nil
+	}
+	return &r.Data
+}
+
+func getEpisode(id string) *episode {
+	return get[episode](`
+query getEpisode($id: ID!) {
+    episode(id: $id) {
+        title
+        description
+        image
+        number
+        season {
+            title
+            show {
+                title
+            }
+        }
+    }
+}
```

Comment:
Can we somehow easily have this in a file and "import" it into go at compile time? 
I just feel it's much nicer if the queries can be edited separately from the res of the code. 

This feels like it could be a super easy solution: https://pkg.go.dev/embed https://blog.jetbrains.com/go/2021/06/09/how-to-use-go-embed-in-go-1-16/